src/rulesets/Base/base.jl

```diff
@@ -75,6 +75,32 @@ function rrule(::typeof(hypot), z::Complex)
     return (Ω, hypot_pullback)
 end
 
+# `sin`
+
+function rrule(::typeof(sin), x::Number)
+    sinx, cosx = sincos(x)
+    sin_pullback(dy) = (NO_FIELDS, cosx * dy)
```

Comment:
Could you use the notation for inputs to pushforwards/pullbacks documented [here](https://www.juliadiff.org/ChainRulesCore.jl/dev/FAQ.html#What-is-up-with-the-different-symbols?)?
```suggestion
    sin_pullback(Δy) = (NO_FIELDS, cosx * Δy)
```